### PR TITLE
[FSDE-936] CHI DevOps GCR Integration

### PR DESCRIPTION
**Pull request by julianatienza**

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/5/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://3bba75c9131c9a7c45a2be530c69fb7751e9420c-chi-pr.enterprise.ctl.io

**Review by jorgedelaflor — APPROVED** [review]
+1

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/6/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/7/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://9f6dd4e26ed753870c8aee20f4c9e7b04e6a3cb4-chi-pr.enterprise.ctl.io

File: cicd/docker/jira/Dockerfile [hunk 1]
```diff
@@ -15,10 +15,15 @@ RUN apt-get -y update && \
 RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
 RUN pip3 install attrdict==2.0.1 pysnow==0.7.17 jsonschema==3.2.0 nose==1.3.7 mock==4.0.3 coverage==5.5
 
-ARG ACLI=atlassian-cli-9.3.0
-ADD https://bobswift.atlassian.net/wiki/download/attachments/16285777/atlassian-cli-9.3.0-distribution.zip /opt/${ACLI}.zip
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
+RUN pip3 install attrdict==2.0.1 pysnow==0.7.17 jsonschema==3.2.0 nose==1.3.7 mock==4.0.3 coverage==5.5
```
Comment (jfranciscohernbal) [inline cicd/docker/jira/Dockerfile:19]: Duped lines in this dockerfile!

**Review by jfranciscohernbal — APPROVED** [review]
Just remove the duplicate lines from JIRA Dockerfile!

**Review by jfranciscohernbal — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/8/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/9/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://a815ef114422f99f909a77d2f2dd942bbbef3f60-chi-pr.enterprise.ctl.io

**Comment by AIRossow** [comment]
I think the Jira Deployment Info calls look ok. Just wanted to show an example of another way the Jira Deployment could be added. 

The three places that are determining the environment could be grouped into one function similar to this: https://github.com/AIRossow/po-enterprise-shell/blob/FSDE-622/cicd/jenkins/Jenkinsfile#L976-L991 and that is used in the post section: https://github.com/AIRossow/po-enterprise-shell/blob/FSDE-622/cicd/jenkins/Jenkinsfile#L804-L806. I think it's good either way, but it might make it easier if the team decides to change their environment naming convention in the future.

**Comment by julianatienza** [comment]
> I think the Jira Deployment Info calls look ok. Just wanted to show an example of another way the Jira Deployment could be added.
> 
> The three places that are determining the environment could be grouped into one function similar to this: https://github.com/AIRossow/po-enterprise-shell/blob/FSDE-622/cicd/jenkins/Jenkinsfile#L976-L991 and that is used in the post section: https://github.com/AIRossow/po-enterprise-shell/blob/FSDE-622/cicd/jenkins/Jenkinsfile#L804-L806. I think it's good either way, but it might make it easier if the team decides to change their environment naming convention in the future.

Thanks @AIRossow . I think that I will keep my approach because in this case, they hadn't pre-pro environment (this team only merge to master and generate release when there is a version bump).

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/10/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://620394dd9e0b4b55fba733ddc18a07a708a052e9-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/11/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://620394dd9e0b4b55fba733ddc18a07a708a052e9-chi-pr.enterprise.ctl.io

**Comment by jorgedelaflor** [comment]
> > I think the Jira Deployment Info calls look ok. Just wanted to show an example of another way the Jira Deployment could be added.
> > The three places that are determining the environment could be grouped into one function similar to this: https://github.com/AIRossow/po-enterprise-shell/blob/FSDE-622/cicd/jenkins/Jenkinsfile#L976-L991 and that is used in the post section: https://github.com/AIRossow/po-enterprise-shell/blob/FSDE-622/cicd/jenkins/Jenkinsfile#L804-L806. I think it's good either way, but it might make it easier if the team decides to change their environment naming convention in the future.
> 
> Thanks @AIRossow . I think that I will keep my approach because in this case, they hadn't pre-pro environment (this team only merge to master and generate release when there is a version bump).

The only side effect I see in creating/updating the JIRA deployment in the post->always section is that that piece of code will be executed 'after' the adoption stats stage, and therefore will never be able to be included in the stats.

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/12/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/13/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/14/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://ac77f6f4af7bdd4562570e622a147c777e57d34a-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/15/. :white_check_mark:

**Comment by AIRossow** [comment]
> > I think the Jira Deployment Info calls look ok. Just wanted to show an example of another way the Jira Deployment could be added.
> > The three places that are determining the environment could be grouped into one function similar to this: https://github.com/AIRossow/po-enterprise-shell/blob/FSDE-622/cicd/jenkins/Jenkinsfile#L976-L991 and that is used in the post section: https://github.com/AIRossow/po-enterprise-shell/blob/FSDE-622/cicd/jenkins/Jenkinsfile#L804-L806. I think it's good either way, but it might make it easier if the team decides to change their environment naming convention in the future.
> 
> Thanks @AIRossow . I think that I will keep my approach because in this case, they hadn't pre-pro environment (this team only merge to master and generate release when there is a version bump).

Ok. That makes sense. I think it looks good.

**Review by AIRossow — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/16/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://47380621b82528d7d5141ed2d54a52e54f511b6e-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/17/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/18/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://44f9a354e416b7179355f279de45c4b143466cfb-chi-pr.enterprise.ctl.io

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-103/19/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://44f9a354e416b7179355f279de45c4b143466cfb-chi-pr.enterprise.ctl.io

**Review by janibolkvadze — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Pull request merged**